testing_and_setup/compass/ocean/jigsaw_to_MPAS/inject_meshDensity.py

```diff
@@ -16,48 +16,63 @@
 import sys
 
 
-def inject_meshDensity(cw_filename, mesh_filename):
+def inject_meshDensity(cw_filename, mesh_filename, on_sphere=True):
     print('Read cell width field from nc file regular grid...')
     ds = nc4.Dataset(cw_filename,'r')
     cellWidth = ds.variables['cellWidth'][:]
-    lon = ds.variables['lon'][:]
-    lat = ds.variables['lat'][:]
+    if on_sphere:
+       lon = ds.variables['lon'][:]
+       lat = ds.variables['lat'][:]
+    else:
+       x = ds.variables['x'][:]
+       y = ds.variables['y'][:]
     ds.close()
 
-    # Add extra column in longitude to interpolate over the Date Line
-    cellWidth = np.concatenate(
-        (cellWidth, cellWidth[:, 0:1]), axis=1)
-    LonPos = np.deg2rad(np.concatenate(
-        (lon.T, lon.T[0:1] + 360)))
-    LatPos = np.deg2rad(lat.T)
-    # set max lat position to be exactly at North Pole to avoid interpolation
-    # errors
-    LatPos[np.argmax(LatPos)] = np.pi / 2.0
+    if on_sphere:
+       # Add extra column in longitude to interpolate over the Date Line
+       cellWidth = np.concatenate(
+           (cellWidth, cellWidth[:, 0:1]), axis=1)
+       LonPos = np.deg2rad(np.concatenate(
+           (lon.T, lon.T[0:1] + 360)))
+       LatPos = np.deg2rad(lat.T)
+       # set max lat position to be exactly at North Pole to avoid interpolation
+       # errors
+       LatPos[np.argmax(LatPos)] = np.pi / 2.0
     minCellWidth = cellWidth.min()
-    meshDensityVsLatLon = (minCellWidth / cellWidth)**4
+    meshDensityVsXY = (minCellWidth / cellWidth)**4
     print('  minimum cell width in grid definition:', minCellWidth)
     print('  maximum cell width in grid definition:', cellWidth.max())
 
-    LON, LAT = np.meshgrid(LonPos, LatPos)
+    if on_sphere:
+       X, Y = np.meshgrid(LonPos, LatPos)
+    else:
+       X, Y = np.meshgrid(x, y)
 
     print('Open unstructured MPAS mesh file...')
     ds = nc4.Dataset(mesh_filename, 'r+')
     meshDensity = ds.variables['meshDensity']
 
-    print('Preparing interpolation of meshDensity from lat/lon to mesh...')
+    print('Preparing interpolation of meshDensity from native coordinates to mesh...')
     meshDensityInterp = interpolate.LinearNDInterpolator(
-        np.vstack((LAT.ravel(), LON.ravel())).T, meshDensityVsLatLon.ravel())
+        np.vstack((X.ravel(), Y.ravel())).T, meshDensityVsXY.ravel())
 
     print('Interpolating and writing meshDensity...')
-    meshDensity[:] = meshDensityInterp(
-        np.vstack(
-            (ds.variables['latCell'][:],
+    if on_sphere:
+       meshDensity[:] = meshDensityInterp(
+           np.vstack(
+            (ds.variables['lonCell'][:],
              np.mod(
-                ds.variables['lonCell'][:] +
+                ds.variables['latCell'][:] +
                 np.pi,
                 2 *
                 np.pi) -
                 np.pi)).T)
```

Comment:
It seems like there's an ocean-related change here. I feel like the order should have been `np.vstack((ds.variables['lonCell'][:], ds.variables['lonCell'][:]))` like it is now, but the `np.mod()` should still operate on `ds.variables['lonCell'][:]` (I think). @mark-petersen, do you have an opinion here?

Reply:
@sbrus89 , to be clear, I don't think I changed the meaning of this operation.  The only thing that changed from the original version was the order of latCell and lonCell was flipped to match the new convention I introduced of X,Y for both planar and sphere coordinates.  Here is what the two versions of the code look like without the confusing diff format:

Old:
```
    meshDensity[:] = meshDensityInterp(
        np.vstack(
            (ds.variables['latCell'][:],
             np.mod(
                ds.variables['lonCell'][:] +
                np.pi,
                2 *
                np.pi) -
                np.pi)).T)
```

New:
```
       meshDensity[:] = meshDensityInterp(
           np.vstack(
            (ds.variables['lonCell'][:],
             np.mod(
                ds.variables['latCell'][:] +
                np.pi,
                2 *
                np.pi) -
                np.pi)).T)
```

Ok, now that I write it out, it looks you are right that the mod operation and subsequent subtraction of pi needs to stay with lonCell.  An alternative would be to use Y,X format for both sphere and planar meshes (and keep this confusing operation unchanged).  Let me know what you prefer and I can update the PR.

Reply:
@matthewhoffman, I probably prefer the X,Y convention as you have it (as long as the mod() goes with lonCell like you mentioned). However, I know @mark-petersen wrote this so it's possible he has a reason to keep it Y,X. I think as long as the order of the coordinates used in setting up the interpolation object are consistent with the order with which it is called, it shouldn't matter. Personally, X,Y is more intuitive.

Reply:
@sbrus89 , I've pushed a commit that corrects the arguments in the call to meshDensityInterp.

Reply:
Looks good to me!